README.md

```diff
@@ -1 +1,128 @@
+yielding messages as they are available. `consume` method accepts two parameters: `timeout` and `maxBytes`.
+`timeout` is the maximum time the consumer will wait for the messages. `maxBytes` is the maximum size of the messages
+to fetch in a single request. Both of these settings are complementary to the settings in `ConsumerOptions` and to the
+server settings (check the Kafka documentation for more information).
+
+If you didn't set the consumer to auto-commit messages then you should use consumers `commit` method to do so. Simply
+pass it a message you wish to commit. For most cases it's recommended to leave auto-commit off and manually commit
+each message as you process them.
+
+```php
+$consumerFactory = new ConsumerFactory($restClient);
+$consumer = $consumerFactory->create('your-consumer-group', Subscription::topic('your-topic'));
+foreach ($consumer->consume() as $message) {
+    // Do your magic
+    $logger->info('got new message', $message->content);
+
+    // ... and when you are done, commit the message.
+    $consumer->commit($message);
+}
+```
+
+### BatchConsumer
+
+BatchConsumer works the same way as Consumer does; the difference is that BatchConsumer doesn't yield each message
+separately but first puts them in batches (`MessagesBatch`). These batches can be configured to be "limited" by either
+count of messages in the batch (`maxCount`), by time (`maxDuration`) or by both (setting both `maxCount` and
+`maxDuration`). If you set `maxDuration` then the batch won't ever take longer than that (+ few ms for processing) as
+it changes the `timeout` parameter of the consumer (consumer won't get stuck on network waiting for more messages).
+
+BatchConsumer can be quite useful for processing of "large" data sets, where you would have to otherwise batch
+the messages yourself (eg. for inserting into database, where batch operations are always better).
+
+As mention in the Consumer example, you may need to commit the messages. For that there is `commit` method,
+which accepts the yielded `MessagesBatch` and commits all the messages inside it in one request.
+
+```php
+$batchConsumerFactory = new BatchConsumerFactory($restClient);
+$batchConsumer = $batchConsumerFactory->create(
+    'your-consumer-group',
+    Subscription::topic('your-topic'),
+    $maxCount = 10000, // Yield the batch when there is 10 000 messages in it
+    $maxDuration = 60  // or when 60 seconds passes, whichever comes first.
+);
+foreach ($batchConsumer->consume() as $messagesBatch) {
+    // The batch might be empty if you specified the maxDuration.
+    if ($messagesBatch->isEmpty()) {
+        continue;
+    }
+
+    // Do your magic
+    $database->insert($messagesBatch->getMessages());
+
+    // ... and when you are done, commit the batch.
+    $batchConsumer->commit($messagesBatch);
+}
+```
+
+Notes
+-----
+
+If there is an error inside `consume` method, I would recommended throwing that consumer out and creating a new one.
```

Comment:
how can I throw consumer out? :D

Reply:
don't use it anymore, unset, let it be GCed ... how would you reword it?

Reply:
Maybe just "create new one" is enough?

Reply:
`I would recommend not using it anymore and creating a new one`?

Reply:
`to recreate the consumer` mozna?